business/mesh.go

```diff
@@ -545,3 +552,36 @@ func (in *MeshService) resolveNetwork(clusterName string, kubeconfig *kubernetes
 
 	return networkName
 }
+
+// GetMeshConfig obtains the mesh configuration from Istio configmap
+func (in *MeshService) GetMeshConfig() (*MeshConfig, error) {
+	cfg := config.Get()
+
+	istioConfigMap, err := in.k8s.GetConfigMap(cfg.IstioNamespace, cfg.ExternalServices.Istio.ConfigMapName)
+	if err != nil {
+		return nil, err
+	}
+
+	meshConfig := MeshConfig{}
+	err = yaml.Unmarshal([]byte(istioConfigMap.Data["mesh"]), &meshConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &meshConfig, nil
+}
+
+// GetRootNamespace returns the root namespace if configured or the control plane namespace
+func (in *MeshService) GetRootNamespace() string {
+	meshConfig, err := in.GetMeshConfig()
+
+	if err == nil && meshConfig.RootNamespace != "" {
+		return meshConfig.RootNamespace
+	}
+	return config.Get().IstioNamespace
+}
+
+// IsRootNamespace returns true if the namespace is equal to the root namespace
+func (in *MeshService) IsRootNamespace(namespace string) bool {
+	return namespace == in.GetRootNamespace()
```

Comment:
I addressed all the review comments including accessing the cache first to get the CM, but as @xeviknal noticed, there is the unmarshaling time that we cannot avoid unless we store the string directly into the cache. 

So, I worked in the solution to get the root namespace directly from the Kiali config:

Operator PR: https://github.com/kiali/kiali-operator/pull/441
Helm Chart PR: https://github.com/kiali/helm-charts/pull/114

For me, it would be better to get the value from Istio CM, but it's true that it usually will be "istio-system" (the default), so being configured in Kiali is not bad. If we follow this path, we need to document where this value is configured and if it is changes in Istio, Kiali will need to be updated too.